Die neusete Stadn ist jetzt in der haiopy.py zusammengefasst

koyi8 opened the pull request:
wie oben

https://github.com/pyfar/haiopy/commit/5ea6457ef2d12c0ed8b2df51d32a4f546a782bc1

koyi8 committed:
test_test

koyi8 committed:
test2

koyi8 committed:
Hello, hier ist das update

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+        _AudioIO.__init__(self, blocksize, buffersize)  
+
+        self.blocksize = blocksize 
+        self.buffersize = buffersize 
```
I think you don't need to do this here. Its already done in _AudioIO.__init__

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+        self.buffersize = buffersize 
+
+        # attribute:
+        self.audio_in = audio_in 
```
This one seem to miss a property and setter :)

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+    @device_out.setter 
+    def device_out(self,value):
+        """Set Index of Output Device"""
+        self._device_out = int(value)
```
It might be good to check, if the device exists.

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+    @channels_out.setter
+    def channels_out(self,value):
+        """ Set number of Output Channels """
+        self._channels_out = value
```
It might be good to check if the channels exist.

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+        else:
+            self.filename = self.audio_in
+
+            self.type_in = 'wav'
```
I would suggest to save audio_in in a single private variable self._audio_in to make the class structure easier.

For the wav file, you could check if audio_in ends with wav and if the file exists

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+            self.filename = self.audio_in
+
+            self.type_in = 'wav'
+            #raise TypeError('audio_in must be of type Signal or String(Wav)')
```
This should definitely be raised :)

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+
+    def play(self):
+
+        self.check_audio_in()
```
I would also do that check in the init to give direct feedback when the Object is created.

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+                data = self.signal.time[read_idx:read_idx + self.blocksize]
+                                
+                if data.size == 0:
+                    break
```
This looks like the code will crash, e.g., if you want to read 2048 samples but there are only 1000 samples left in the signal

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+
+            self.stream.start()
+
+            self.timeout = self.blocksize * self.buffersize / self.stream.samplerate
```
This code is identical for both methds. You should put it in a separate private class method :)

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+    stream = None
+
+    def __init__(self, 
+                audio_out,
```
- should be audio_in :)
- it might be better to pass either 'signal' or the name of the wav file.

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+                blocksize=2048,
+                buffersize=20,
+                device_out = 1,
+                channels_out = 2): 
```
A view thoughts on the channel handling in general:
- It might be convenient to have channels_out=None as the default. This would than select the first N channels to play the N channels of audio_in
- Passing in single channel, will select channels `np.arange(channels_out, channels_out + N)` where N are the number of channels from audio_out.
- Passing an array like (list, or np.array) will select the exact channels to play back the audio

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+    """ Class for Playback of .wav or Signal-Objects with chosen Sounddevice """
+
+    def __init__(self, 
+                audio_in, # isinstance entweder wav string oder pyfar signal type
```
should be audio_out :)

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+                audio_out,
+                blocksize=2048,
+                buffersize=20,
+                device_in=0,
```
- the default for device_out in the Playback class is `1`. Should be consistent across classes.
- I think channels_in is missing

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+   
+
+
+class Recording(_AudioIO):
```
Lets call this `Record` to be consistent with `Play`

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+                blocksize=2048,
+                buffersize=20,
+                device_in=0,
+                device_out=1):
```
misses channels_in and channels_out

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+
+
+
+        self.device_in = device_in
+        self.device_out = device_out
+        self.audio_in = audio_in
+        self.audio_out = audio_out
+        self.blocksize = blocksize
+        self.buffersize = buffersize
```
This should not be needed - it is already done when you call the init functions of the inherited classes

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+        
+
+
+class PlayRec(Play,Recording):
```
Lets name this one PlayRecord :)

f-brinkmann commented on haiopy/haiopy.py:
```diff
@@ -1 +1,590 @@
+        
+
+        if status:
+            print(status)
```
what is printed here? I think we should limit the prints to the command line to errors and serious warnings...

f-brinkmann reviewed (changes requested):
This is a good step in the right direction. I made my comments below :)

richarddeffner committed:
Add files via upload

Hi, hier ein update vor dem Wochenende:
- non-blocking recording
- samplerate / dtype / channel checks
- flake8
